frappe/email/doctype/newsletter/newsletter.js

```diff
@@ -4,23 +4,60 @@
 frappe.ui.form.on('Newsletter', {
 	refresh(frm) {
 		let doc = frm.doc;
-		if(!doc.__islocal && !cint(doc.email_sent) && !doc.__unsaved
+		if (!doc.__islocal && !cint(doc.email_sent) && !doc.__unsaved
 				&& in_list(frappe.boot.user.can_write, doc.doctype)) {
-			frm.add_custom_button(__('Send'), function() {
-				frm.call('send_emails').then(() => {
-					frm.refresh();
+			frm.add_custom_button(__('Send Now'), function() {
+				frappe.confirm(__("Do you really want to send this email newsletter?"), function() {
+					frm.call('send_emails').then(() => {
+						frm.set_value('schedule_send', new Date());
+						frm.save();
+					});
 				});
 			}, "fa fa-play", "btn-success");
 		}
+		if (!doc.__islocal && cint(doc.email_sent)) {
+			frm.set_df_property('schedule_send', "read_only", 1);
+		}
 
 		frm.events.setup_dashboard(frm);
 
-		if(doc.__islocal && !doc.send_from) {
+		if (doc.__islocal && !doc.send_from) {
 			let { fullname, email } = frappe.user_info(doc.owner);
 			frm.set_value('send_from', `${fullname} <${email}>`);
 		}
 	},
 
+	onload_post_render(frm) {
+		frm.trigger('setup_schedule_send');
+	},
+
+	schedule_send(frm) {
+		frm.trigger('setup_schedule_send');
+	},
+
+	setup_schedule_send(frm) {
```

Comment:
- Schedule Send value is set to empty on saving
![img1](https://user-images.githubusercontent.com/7310479/77881570-e7e0c500-727c-11ea-8cc9-d75720748e8b.gif)